### ISSUE TITLE
SwiftFormat creating incorrect SwiftData code: "A default value requires a fully qualified domain named value"

### ISSUE THREAD

**Issue by glacials** [nicklockwood/SwiftFormat#1649]
Thanks for your hard work on SwiftFormat! This issue is not urgent and feel free to wontfix, but wanted to get it logged.

# Issue

I have this code:

```swift
import Foundation
import SwiftData

@Model
class FooBar {
  var happenedAt: Date = Date.now
  init() {}
}
```

Running SwiftFormat makes this change:

```diff
 import Foundation
 import SwiftData

 @Model
 class FooBar {
-  var happenedAt: Date = Date.now
+  var happenedAt: Date = .now
   init() {}
 }
```

which yields this error on the affected line:

> A default value requires a fully qualified domain named value

This only happens on `@Model` classes.

# Workaround

To work around the issue, I can remove the type hint and put the fully qualified domain named value back in, which makes SwiftFormat leave it alone:

```diff
 import Foundation
 import SwiftData

 @Model
 class FooBar {
-  var happenedAt: Date = Date.now
+  var happenedAt = Date.now
   init() {}
 }
```

Thanks for reading! SwiftFormat is the best.

**Comment by nicklockwood** [comment]
@glacials thanks for flagging this, I'll get it fixed. The rule causing the issue is [redundantType](https://github.com/nicklockwood/SwiftFormat/blob/main/Rules.md#redundanttype).

You can use the option `--redundanttype inferred` to avoid this problem for now as it basically does the same thing as your workaround.

**Comment by nicklockwood** [comment]
@glacials fixed in 0.53.6

**Comment by glacials** [comment]
Thank you!!
